pipe_tools/utils/flatten.py

```diff
@@ -0,0 +1,35 @@
+import six
+
+
+def flatten(struct):
+    """
+    Creates a flat list of all all items in structured output (dicts, lists, items):
+    .. code-block:: python
+        >>> sorted(flatten({'a': 'foo', 'b': 'bar'}))
+        ['bar', 'foo']
+        >>> sorted(flatten(['foo', ['bar', 'troll']]))
+        ['bar', 'foo', 'troll']
+        >>> flatten('foo')
+        ['foo']
+        >>> flatten(42)
+        [42]
+    """
+    if struct is None:
+        return []
```

Comment:
I'd probably write the stuff below here using list comprehensions:

        if isinstance(struct, dict):
           return [flatten(x) for x in six.itervalues(struct)]
        if isinstance(struct, six.string_types):
           return [struct]
   
        try:
            return [flatten(x) for x in struct]
        except:
            return [struct]

But that's a matter of taste.